crates/hyperdrive-math/src/lib.rs

```diff
@@ -124,7 +124,7 @@ impl State {
         let annualized_time =
             self.position_duration() / FixedPoint::from(U256::from(60 * 60 * 24 * 365));
         let spot_price = self.calculate_spot_price();
-        (fixed!(1e18) - spot_price) / (spot_price * annualized_time)
+        calculate_fixed_rate_from_price(spot_price, annualized_time)
```

Comment:
This is nice, but I was thinking something more along the lines of:

```rs
pub fn calculate_spot_rate(&self, spot_price: Option<FixedPoint>) -> FixedPoint {
      let annualized_time =
          self.position_duration() / FixedPoint::from(U256::from(60 * 60 * 24 * 365));
      let spot_price = match spot_price {
          Some(spot_price) => spot_price,
          None => self.calculate_spot_price(),
      };
      (fixed!(1e18) - spot_price) / (spot_price * annualized_time)
  }
```

This way we could easily convert with `self.calculate_spot_rate(arbitrary_spot_price)`